### PR TITLE
feat(strict-ts): apply to common utils

### PR DESCRIPTION
**Pull request by capJavert**
daily.dev strict mode PR template

This is the one of the PRs in a series to bring our TypeScript types to comply with [strict](https://www.typescriptlang.org/tsconfig/#strict) config option. This will allow us better type checking during builds and development and allow us to ship with more confidence.

We are doing this as a series of PRs to have easier review and also to incrementally deploy to production.

This one focuses on:
- common
- top level utils

### CODE REVIEW

File: src/routes/rss.ts [hunk 1]
```diff
@@ -90,7 +91,9 @@ const generateRSS =
         }),
       );
       return res.type('application/rss+xml').send(feed.xml());
-    } catch (err) {
+    } catch (originalError) {
+      const err = originalError as TypeORMQueryFailedError;
+
```
Comment (omBratteng) [inline src/routes/rss.ts:96]: I really wish we could set type inside `catch` statement :triggered:

**Review by omBratteng — APPROVED** [review]

File: src/routes/rss.ts (hunk 1 above)
Comment (rebelchris) [reply src/routes/rss.ts:96]: Yep really sad :(

File: src/common/object.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ export const isNullOrUndefined = (value: unknown) =>
   typeof value === 'undefined' || value === null;
 
 type Key = string;
-type Value = string;
+type Value = string | undefined;
```
Comment (rebelchris) [inline src/common/object.ts:21]: Wouldnt this do?
```suggestion
type Value = string?;
```

**Review by rebelchris — APPROVED** [review]
For the undefined ones can't we just make things optional?
I think it's slightly cleaner as you don't have to parse `as string` everywhere.

File: src/common/object.ts (hunk 2 above)
Comment (rebelchris) [reply src/common/object.ts:21]: Oh sigh 😮‍💨

**Pull request merged**